### PR TITLE
feat: add RDMA definition to datasource

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: add RDMA definition to datasource

#### Which issue(s) this PR fixes

Fix # https://github.com/kubeagi/arcadia/issues/442

### CODE REVIEW

File: api/base/v1alpha1/datasource_types.go [hunk 1]
```diff
@@ -32,6 +32,19 @@ type DatasourceSpec struct {
...
+}
+
+type RDMA struct {
+	// Path on a model storage server, the usual storage path is /path/ns/mode-name, and the path field is /path/, which must end in /.
+	// example: /opt/paas/, /opt/, /
+	// +kubebuilder:validation:Pattern=(^\/$)|(^\/[a-zA-Z0-9\_.@-]+(\/[a-zA-Z0-9\_.@-]+)*\/$)
+	Path string `json:"path"`
+
+	// Host trans-core's service address
+	Host string `json:"host"`
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:47]: We use  `endpoint` to set this Host

File: api/base/v1alpha1/datasource_types.go [hunk 2]
```diff
@@ -32,6 +32,19 @@ type DatasourceSpec struct {
...
 	// OSS defines info for object storage service
 	OSS *OSS `json:"oss,omitempty"`
+
+	// RDMA configure RDMA pulls the model file directly from the remote service to the host node.
+	RDMA *RDMA `json:"rdma,omitempty"`
+}
+
+type RDMA struct {
+	// Path on a model storage server, the usual storage path is /path/ns/mode-name, and the path field is /path/, which must end in /.
+	// example: /opt/paas/, /opt/, /
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:42]: /opt/kubeagi

File: controllers/model_controller.go [hunk 3]
```diff
@@ -174,34 +175,40 @@ func (r *ModelReconciler) Initialize(ctx context.Context, logger logr.Logger, in
...
-	}
+	var (
+		ds   datasource.Datasource
+		info any
+	)
+
+	// If source is empty, it means that the data is still sourced from the internal minio and a state check is required,
+	// otherwise we consider the model file for the trans-core service to be ready.
+	if instance.Spec.Source == nil {
+		klog.V(5).Infof("model %s source is empty, check minio status.", instance.Name)
```
Comment (Abirdcfly) [inline controllers/model_controller.go:187]: use logger in function arg please.

**Review by Abirdcfly — APPROVED** [review]

File: api/base/v1alpha1/datasource_types.go [hunk 4]
```diff
@@ -32,6 +32,19 @@ type DatasourceSpec struct {
...
+}
+
+type RDMA struct {
+	// Path on a model storage server, the usual storage path is /path/ns/mode-name, and the path field is /path/, which must end in /.
+	// example: /opt/kubeagi/, /opt/, /
+	// +kubebuilder:validation:Pattern=(^\/$)|(^\/[a-zA-Z0-9\_.@-]+(\/[a-zA-Z0-9\_.@-]+)*\/$)
+	Path string `json:"path"`
+
+	// Endpoint trans-core's service address. Just need ip or dns. No protocols required
+	Endpoint string `json:"host"`
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:47]: well.. I mean we have a field in DatasourceSpec  https://github.com/kubeagi/arcadia/blob/main/api/base/v1alpha1/datasource_types.go#L31

**Review by bjwswang — APPROVED** [review]

**Pull request merged**